share/wake/lib/core/list.wake

```diff
@@ -179,63 +267,100 @@ export def scanr (combiningFn: element => accumulator => accumulator): accumulat
 # Concatenate two lists into one List.
 # The ++ operator must be provided by the wake package for publish to work.
 #
+# Guarantees:
+#  - ++ is associtive
+#  - Nil is an identity of ++
```

Comment:
In contrast, this associative guarantee is super relevant. It might help to write `x ++ (y ++ z) == (x ++ y) ++ z`, though, for people who don't know what that word means.